bambi/backend/pymc.py

```diff
@@ -23,6 +24,23 @@
 __version__ = version("bambi")
 
 
+PYMC_SAMPLERS = ["mcmc"]
+BAYEUX_SAMPLERS = [
```

Comment:
We could do something like:

```
# TODO(colcarroll): Make this public
import importlib

def get_bayeux_samplers():
  bx_samplers = []
  if importlib.util.find_spec('bayeux') is None:
    return bx_samplers
  import bayeux as bx
  for module in bx._src.bayeux._MODULES:
    mname = module.__name__.split('.')[-1]
    for k in module.__all__:
      bx_samplers.append(mname + '.' + getattr(module, k).name)
  return bx_samplers
```

or, once a version of this is in `bayeux`, 

```
def get_bayeux_samplers():
  if importlib.util.find_spec('bayeux') is None:
    return bx_samplers
  import bayeux as bx
  return bx.algorithms 
```

this currently looks like 

```
['mcmc.tfp_hmc',
 'mcmc.tfp_nuts',
 'mcmc.tfp_snaper_hmc',
 'mcmc.blackjax_hmc',
 'mcmc.blackjax_chees_hmc',
 'mcmc.blackjax_meads_hmc',
 'mcmc.blackjax_nuts',
 'mcmc.blackjax_hmc_pathfinder',
 'mcmc.blackjax_nuts_pathfinder',
 'mcmc.flowmc_rqspline_hmc',
 'mcmc.flowmc_rqspline_mala',
 'mcmc.flowmc_realnvp_hmc',
 'mcmc.flowmc_realnvp_mala',
 'mcmc.numpyro_hmc',
 'mcmc.numpyro_nuts',
 'optimize.jaxopt_bfgs',
 'optimize.jaxopt_gradient_descent',
 'optimize.jaxopt_lbfgs',
 'optimize.jaxopt_nonlinear_cg',
 'optimize.optimistix_bfgs',
 'optimize.optimistix_chord',
 'optimize.optimistix_dogleg',
 'optimize.optimistix_gauss_newton',
 'optimize.optimistix_indirect_levenberg_marquardt',
 'optimize.optimistix_levenberg_marquardt',
 'optimize.optimistix_nelder_mead',
 'optimize.optimistix_newton',
 'optimize.optimistix_nonlinear_cg',
 'optimize.optax_adabelief',
 'optimize.optax_adafactor',
 'optimize.optax_adagrad',
 'optimize.optax_adam',
 'optimize.optax_adamw',
 'optimize.optax_adamax',
 'optimize.optax_amsgrad',
 'optimize.optax_fromage',
 'optimize.optax_lamb',
 'optimize.optax_lion',
 'optimize.optax_noisy_sgd',
 'optimize.optax_novograd',
 'optimize.optax_radam',
 'optimize.optax_rmsprop',
 'optimize.optax_sgd',
 'optimize.optax_sm3',
 'optimize.optax_yogi',
 'vi.tfp_factored_surrogate_posterior']
```

We can then use [`operator.attrgetter`](https://docs.python.org/3/library/operator.html#operator.attrgetter) where you use `getattr` on line 232. 

Note that the benefits of having this global constant here is just for debugging: a user can see valid inputs. You could also just use `pymc` if the input is "mcmc", or _try_ to use `bayeux` and fail in a reasonable way. 

Note also that `bayeux` includes optional dependencies using the same `importlib.util.find_spec` trick, so `flowMC` is not actually required, and the list of available algorithms depends on the user's environment.